web/src/app/schedules/temp-sched/TempSchedShiftsList.tsx

```diff
@@ -132,15 +249,80 @@ export default function TempSchedShiftsList({
           ),
         })
 
+        // signify that a portion of the shift is now added to the list
+        // prevents actions from rendering on subsequent list items for the same shift timespan
         s.added = true
+
+        // check coverage until the next shift (if there is one) within the current day
+        if (
+          s.isValid &&
+          shiftIdx < dayShifts.length - 1 &&
+          checkCoverage(s.end, dayShifts[shiftIdx + 1].start)
+        ) {
+          result.push({
+            render: () => (
+              <Alert
+                key={s.end.millisecond + '-no-middle-coverage'}
+                className={classes.alert}
+                severity='warning'
+              >
+                {`No coverage from ${s.end.setZone(zone).toFormat('h:mm a')} to 
+                ${dayShifts[shiftIdx + 1].start
+                  .setZone(zone)
+                  .toFormat('h:mm a')}`}
+              </Alert>
+            ),
+          })
+        }
+
+        // check end of day/temp sched coverage
+        // if on the day of temp sched's end, temp sched end is used
+        let dayEnd = dayInterval.end
+        if (
+          dayIdx === days.length - 1 &&
+          lastShiftEnd.day === schedInterval.end.day
+        ) {
+          dayEnd = schedInterval.end
+        }
+        if (
+          s.isValid &&
+          shiftIdx === dayShifts.length - 1 &&
+          checkCoverage(s.end, dayEnd)
+        ) {
+          result.push({
+            render: () => (
+              <Alert
+                key={s.end.millisecond + '-no-end-coverage'}
+                className={classes.alert}
+                severity='warning'
+              >
+                No coverage after {s.end.setZone(zone).toFormat('h:mm a')}
```

Comment:
I think it makes sense to just call the API directly in this case, since the only shortening would mainly come from not needing to type out `setZone` a bunch, as the start var in some cases is still pretty long, i.e. `dayShifts[shiftIdx + 1].start`